### PR TITLE
🛡 Enable `ServiceAccount` token projection

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area security
/kind enhancement
/merge squash

**What this PR does / why we need it**:
When the extension is running on a seed with a gardenlet of at least `v1.37` then `ServiceAccount` token projection is enabled.

**Which issue(s) this PR fixes**:
Part of gardener/gardener#4659
Part of gardener/gardener#4878

**Release note**:

```noteworthy operator
The extension controller uses a projected `ServiceAccount` token in case it runs on a seed with a gardenlet of at least `v1.37` or higher.
```

### CODE REVIEW

File: example/controller-registration.yaml [hunk 1]
```diff
@@ -4,11 +4,13 @@ kind: ControllerRegistration
 metadata:
   name: kupid
 spec:
+  resources:
+  - kind: ClusterPorSchedulingPolicy
+    type: 
   deployment:
     type: helm
-    policy: Always
```
Comment (amshuman-kr) [inline example/controller-registration.yaml:9]: I guess you will still need this.

File: example/controller-registration.yaml [hunk 2]
```diff
@@ -4,11 +4,13 @@ kind: ControllerRegistration
 metadata:
   name: kupid
 spec:
+  resources:
+  - kind: ClusterPorSchedulingPolicy
+    type:
```
Comment (amshuman-kr) [inline example/controller-registration.yaml:9]:
```suggestion
  resources:
  - kind: ClusterPodSchedulingPolicy
    type: 
```

Do you need this? If so, technically, `PodSchedulingPolicy` is also applicable here.

**Review by amshuman-kr — COMMENTED** [review]
Thanks for the PR @rfranzke. LGTM except for minor nits on the example controller-registration.yaml. I used the gardener scripts to auto-generate it but the resources and deployment policies are not generated properly. I am sorry, I didn't get around to automating that part 😅

**Comment by amshuman-kr** [comment]
> Did you manually maintain this in the past?

I used to manually maintain the `resources` and `deployment` `policy` sections because I somehow couldn't get the script to generate those parts [right](https://github.com/gardener/kupid/blob/master/example/controller-registration-template.yaml#L9).

**Comment by rfranzke** [comment]
OK, I have reverted these changes for now.

**Review by shreyas-s-rao — APPROVED** [review]

**Review by ashwani2k — APPROVED** [review]
/lgtm

**Pull request merged**